### PR TITLE
Adds AES-GCM-SIV

### PR DESCRIPTION
**Pull request by celic**
Apparently, GitHub does not allow comments on the raw compare page. We must make a PR in order to provide comments in-line.

### CODE REVIEW

File: src/draft-celi-acvp-block-ciph-00.xml [hunk 1]
```diff
@@ -137,6 +137,7 @@
                     <t>ACVP-AES-CFB128</t>
                     <t>ACVP-AES-CTR</t>
                     <t>ACVP-AES-GCM</t>
+                    <t>ACVP-AES-SIV</t>
```
Comment (celic) [inline src/draft-celi-acvp-block-ciph-00.xml]: I'd vote for "ACVP-AES-GCM-SIV" as I believe "AES-SIV" is a different mode of AES already.

File: src/draft-celi-acvp-block-ciph-00.xml [hunk 2]
```diff
@@ -1031,6 +1032,7 @@ For i = 0 to 399
                 <texttable anchor="property_grid_kw" title="Key-Wrap Block Cipher Algorithm Capabilities Applicability Grid">
                     <ttcol align="left">algorithm</ttcol><ttcol align="left">revision</ttcol><ttcol align="left">direction</ttcol><ttcol align="left">keyLen</ttcol><ttcol align="left">payloadLen</ttcol><ttcol align="left">ivLen</ttcol><ttcol align="left">ivGen</ttcol><ttcol align="left">ivGenMode</ttcol><ttcol align="left">saltGen</ttcol><ttcol align="left">aadLen</ttcol><ttcol align="left">tagLen</ttcol>
                     <c>AES-GCM</c><c>"1.0.0"</c><c>["encrypt", "decrypt"]</c><c>[128, 192, 256]</c><c>{"Min": 0, "Max": 65536, "Inc": any}</c><c>{"Min": 8, "Max": 1024, "Inc": any}</c><c>["internal", "external"]</c><c>["8.2.1", "8.2.2"]</c><c></c><c>{"Min": 0, "Max": 65536, "Inc": any}</c><c>{[32, 64, 96, 104, 112, 120, 128]}</c>
+                    <c>AES-SIV</c><c>"1.0.0"</c><c>["encrypt", "decrypt"]</c><c>[128, 192, 256]</c><c>{"Min": 0, "Max": 65536, "Inc": any}</c><c></c><c></c><c></c><c></c><c>{"Min": 0, "Max": 65536, "Inc": any}</c><c></c>
```
Comment (celic) [inline src/draft-celi-acvp-block-ciph-00.xml]: GCM-SIV operates on a 128 or 256 bit key, so 192 should not be in that list. The plaintext length can only be a multiple of 8 bits (the operations are only defined over bytes). This is the same with AAD length.

Comment (bfussell) [reply src/draft-celi-acvp-block-ciph-00.xml]: OK. 192 came from the fact that openssl implemented 192 in addition to 128 and 256.

**Comment by smuellerDD** [comment]
Am Mittwoch, 6. März 2019, 15:49:17 CET schrieb Barry Fussell:

Hi Barry,

> bfussell commented on this pull request.
> 
> > @@ -137,6 +137,7 @@
> 
>                      <t>ACVP-AES-CFB128</t>
>                      <t>ACVP-AES-CTR</t>
>                      <t>ACVP-AES-GCM</t>
> +                    <t>ACVP-AES-SIV</t>
> 
> Hmmm, that may be the issue and the reason for 192 bit key support. Not sure
> this was clearly stated before, AES-SIV vs AES-GCM-SIV. That will
> definitely be more work since openssl does not support AES-GCM-SIV. Are you
> aware of an open source implementation of that ?

https://commondatastorage.googleapis.com/chromium-boringssl-docs/
aead.h.html#EVP_aead_aes_256_gcm_siv

I do have a boringSSL backend for my ACVP Parser and could give it a try.


Ciao
Stephan

**Comment by atvassilev** [comment]
@smuellerDD That would be great! Thanks for your help.

**Comment by smuellerDD** [comment]
Am Mittwoch, 6. März 2019, 16:46:27 CET schrieb Apostol Vassilev:

Hi Apostol,

> @smuellerDD That would be great! Thanks for your help.

Let me know once the ACVP server gained support for AES-GCM-SIV and I will 
report.

Ciao
Stephan

File: src/draft-celi-acvp-block-ciph-00.xml (hunk 1 above)
Comment (fliphil) [reply src/draft-celi-acvp-block-ciph-00.xml]: Yes I based these changes around the AES-SIV, not AES-GCM-SIV. I will make the appropriate changes according to your comments

**Comment by celic** [comment]
@smuellerDD The intention is to code this up on the server at the IETF 104 Hackathon in Prague. We would like the specification to be agreed upon before then. You are welcome to participate in the Hackathon. If you have any questions on it, feel free to reach out via email.

**Comment by celic** [comment]
The changes look good. I can't approve the PR because I "authored" it (even though none of the commits are my own...) We can merge whenever you are ready.

**Comment by bfussell** [comment]
Got it.

**Comment by smuellerDD** [comment]
Am Mittwoch, 13. März 2019, 16:35:41 CET schrieb Chris Celi:

Hi Chris,

> @smuellerDD The intention is to code this up on the server at the IETF 104
> Hackathon in Prague. We would like the specification to be agreed upon
> before then. You are welcome to participate in the Hackathon. If you have
> any questions on it, feel free to reach out via email.

I guess you had a successful Hackathon.

Do you want me to test AES-GCM-SIV? If so, how would I reach your private 
server?

Ciao
Stephan

**Comment by celic** [comment]
Hi Stephan,

Please send an email to christopher.celi (at) NIST.gov and I'll get you the details. 

Chris

**Pull request merged**